Source/WebGPU/WebGPU/BindGroupLayout.h

```diff
@@ -41,13 +41,9 @@ class Device;
 class BindGroupLayout : public WGPUBindGroupLayoutImpl, public RefCounted<BindGroupLayout> {
     WTF_MAKE_FAST_ALLOCATED;
 public:
-    static Ref<BindGroupLayout> create(id<MTLArgumentEncoder> vertexArgumentEncoder, id<MTLArgumentEncoder> fragmentArgumentEncoder, id<MTLArgumentEncoder> computeArgumentEncoder)
+    static Ref<BindGroupLayout> create(HashMap<uint32_t, WGPUShaderStageFlags>&& stageMapTable, id<MTLArgumentEncoder> vertexArgumentEncoder = nil, id<MTLArgumentEncoder> fragmentArgumentEncoder = nil, id<MTLArgumentEncoder> computeArgumentEncoder = nil)
```

Comment:
Why not pass as an array of `id<MTLArgumentEncoder>`s instead of individually? (And instead of `m_vertexArgumentEncoder`, access would be via `m_argumentEncoder[vertex]`)

Reply:
It is possible but I would need to make `vertex` / `fragment` / `compute` accessible via the BindGroupLayout too, currently they are only defined in BindGroup.mm